l2geth/core/state_transition.go

```diff
@@ -239,50 +222,30 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 		vmerr error
 	)
 
-	if vm.UsingOVM {
-		to := "<nil>"
-		if msg.To() != nil {
-			to = msg.To().Hex()
-		}
-		l1MessageSender := "<nil>"
-		if msg.L1MessageSender() != nil {
-			l1MessageSender = msg.L1MessageSender().Hex()
-		}
-		if st.evm.EthCallSender == nil {
-			log.Debug("Applying transaction", "from", sender.Address().Hex(), "to", to, "nonce", msg.Nonce(), "gasPrice", msg.GasPrice().Uint64(), "gasLimit", msg.Gas(), "value", msg.Value().Uint64(), "l1MessageSender", l1MessageSender, "data", hexutil.Encode(msg.Data()))
-		}
-	}
-
 	if contractCreation {
 		ret, _, st.gas, vmerr = evm.Create(sender, st.data, st.gas, st.value)
 	} else {
 		// Increment the nonce for the next transaction
-		if !vm.UsingOVM {
-			// OVM_DISABLED
-			st.state.SetNonce(msg.From(), st.state.GetNonce(msg.From())+1)
-		}
-
+		st.state.SetNonce(msg.From(), st.state.GetNonce(msg.From())+1)
 		ret, st.gas, vmerr = evm.Call(sender, st.to(), st.data, st.gas, st.value)
 	}
 
 	if vmerr != nil {
+		log.Debug("VM returned with error", "err", vmerr, "ret", hexutil.Encode(ret))
+		// The only possible consensus-error would be if there wasn't
+		// sufficient balance to make the transfer happen. The first
+		// balance transfer may never fail.
 		if vmerr == vm.ErrInsufficientBalance {
 			return nil, 0, false, vmerr
 		}
 	}
 	st.refundGas()
+	st.state.AddBalance(evm.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
```

Comment:
Coinbase needs to be Sequencer fee vault

Reply:
This won't work exactly because this accounts for gas refunds